src/Microsoft.Framework.DependencyInjection.Abstractions/ServiceDescriptor.cs

```diff
@@ -72,6 +72,24 @@ private ServiceDescriptor(Type serviceType, ServiceLifetime lifetime)
         /// <inheritdoc />
         public Func<IServiceProvider, object> ImplementationFactory { get; }
 
+        public Type GetImplementationType()
```

Comment:
Why not just put this in the extensions class as a private method rather than create a new public surface area?